ultimarc/devices/_mappings.py

```diff
@@ -0,0 +1,517 @@
+    "DOWN": 0x51,
+    "UP": 0x52,
+    "NUM": 0x53,
+    "KP /": 0x54,
+    "KP *": 0x55,
+    "KP -": 0x56,
+    "KP +": 0x57,
+    "KP ENTER": 0x58,
+    "KP 1": 0x59,
+    "KP 2": 0x5A,
+    "KP 3": 0x5B,
+    "KP 4": 0x5C,
+    "KP 5": 0x5D,
+    "KP 6": 0x5E,
+    "KP 7": 0x5F,
+    "KP 8": 0x60,
+    "KP 9": 0x61,
+    "KP 0": 0x62,
+    "KP .": 0x63,
+    "\\": 0x64,
+    "NON US \\": 0x64,
+    "APP": 0x65,
+    "KEYBOARD POWER": 0x66,
+    "KP =": 0x67,
+    "CTRL L": 0x70,
+    "SHIFT L": 0x71,
+    "ALT L": 0x72,
+    "WIN L": 0x73,
+    "CTRL R": 0x74,
+    "SHIFT R": 0x75,
+    "ALT R": 0x76,
+    "WIN MENU": 0x77,
+    "MOUSE LEFT DBL CLICK": 0x80,
+    "MOUSE LEFT": 0x81,
+    "MOUSE MIDDLE": 0x82,
+    "MOUSE RIGHT": 0x83,
+    "POWER": 0x88,
+    "SLEEP": 0x89,
+    "WAKE": 0x8A,
+    "VOL UP": 0x8B,
+    "VOL DOWN": 0x8C,
+    "GAMEPAD 1": 0x90,
+    "GAMEPAD 2": 0x91,
+    "GAMEPAD 3": 0x92,
+    "GAMEPAD 4": 0x93,
+    "GAMEPAD 5": 0x94,
+    "GAMEPAD 6": 0x95,
+    "GAMEPAD 7": 0x96,
+    "GAMEPAD 8": 0x97,
+    "GAMEPAD 9": 0x98,
+    "GAMEPAD 10": 0x99,
+    "GAMEPAD 11": 0x9A,
+    "GAMEPAD 12": 0x9B,
+    "GAMEPAD 13": 0x9C,
+    "GAMEPAD 14": 0x9D,
+    "GAMEPAD 15": 0x9E,
+    "GAMEPAD 16": 0x9F,
+    "GAMEPAD 17": 0xA0,
+    "GAMEPAD 18": 0xA1,
+    "GAMEPAD 19": 0xA2,
+    "GAMEPAD 20": 0xA3,
+    "GAMEPAD 21": 0xA4,
+    "GAMEPAD 22": 0xA5,
+    "GAMEPAD 23": 0xA6,
+    "GAMEPAD 24": 0xA7,
+    "GAMEPAD 25": 0xA8,
+    "GAMEPAD 26": 0xA9,
+    "GAMEPAD 27": 0xAA,
+    "GAMEPAD 28": 0xAB,
+    "GAMEPAD 29": 0xAC,
+    "GAMEPAD 30": 0xAD,
+    "GAMEPAD 31": 0xAE,
+    "GAMEPAD 32": 0xAF,
+    "ANALOG 0": 0xB0,
+    "ANALOG 1": 0xB1,
+    "ANALOG 2": 0xB2,
+    "ANALOG 3": 0xB3,
+    "ANALOG 4": 0xB4,
+    "ANALOG 5": 0xB5,
+    "ANALOG 6": 0xB6,
+    "ANALOG 7": 0xB7,
+    "HAT 0": 0xBA,
+    "HAT 1": 0xBB,
+    "HAT 2": 0xDC,
+    "HAT 3": 0xBD,
+    "TRACKBALL X": 0xC0,
+    "TRACKBALL Y": 0xC2,
+    "TRACKBALL Z": 0xC4,
+    "m1": 0xe0,
+    "m2": 0xe1,
+    "m3": 0xe2,
+    "MUTE": 0xe2,
+    "m4": 0xe3,
+    "PLAY/PAUSE": 0xe3,
+    "m5": 0xe4,
+    "NEXT": 0xe4,
+    "m6": 0xe5,
+    "PREV": 0xe5,
+    "m7": 0xe6,
+    "STOP": 0xe6,
+    "m8": 0xe7,
+    "m9": 0xe8,
+    "m10": 0xe9,
+    "m11": 0xea,
+    "m12": 0xeb,
+    "m13": 0xec,
+    "m14": 0xed,
+    "m15": 0xee,
+    "m16": 0xef,
+    "m17": 0xf0,
+    "EMAIL": 0xf0,
+    "m18": 0xf1,
+    "SEARCH": 0xf1,
+    "m19": 0xf2,
+    "BOOKMARKS": 0xf2,
+    "m20": 0xf3,
+    "OPEN BROWSER": 0xf3,
+    "m21": 0xf4,
+    "WEB BACK": 0xf4,
+    "m22": 0xf5,
+    "WEB FORWARD": 0xf5,
+    "m23": 0xf6,
+    "WEB STOP": 0xf6,
+    "m24": 0xf7,
+    "WEB REFRESH": 0xf7,
+    "m25": 0xf8,
+    "MEDIA PLAYER": 0xf8,
+    "m26": 0xf9,
+    "m27": 0xfa,
+    "CALCULATOR": 0xfa,
+    "m28": 0xfb,
+    "m29": 0xfc,
+    "EXPLORER": 0xfc,
+    "m30": 0xfd,
+    "m31": 0xfe,
+    "WAIT 3 SEC": 0xfe,
+}
+
+#
+# Pre 2015 IPAC key mapping
+#
+PREIPACMapping = {
```

Comment:
I would like to use legacy instead of pre when dealing with the older boards.  How do you feel about it?

Reply:
I'm good with that.